File: solution/0600-0699/0651.4 Keys Keyboard/Solution.py
class Solution:
    def maxA(self, n: int) -> int:
        f = [0] * (n + 1)
        for i in range(1, n + 1):
            f[i] = f[i - 1] + 1
            for j in range(2, i):
                f[i] = max(f[i], f[j - 2] * (i - j + 1))
        return f[n]

File: solution/0600-0699/0651.4 Keys Keyboard/Solution.java
class Solution {
    public int maxA(int n) {
        int[] f = new int[n + 1];
        for (int i = 1; i <= n; ++i) {
            f[i] = f[i - 1] + 1;
            for (int j = 2; j < i; ++j) {
                f[i] = Math.max(f[i], f[j - 2] * (i - j + 1));
            }
        }
        return f[n];
    }
}